### PR TITLE
Njones/krakend tls

### PR DESCRIPTION
**Pull request by njones-lanl**
This merge request adds TLS to the frontend of krakend, with automatic renewal via the ACME protocol.

It also adds unit tests to cover TLS certificate generation with ACME, and the health endpoint of the krakend service.

This merge request does include use a custom container for both the ACME provider (step-ca, which didn't have a great container for our use cases), and the ACME client (krakend, which needs the ACME provider loaded inside the container due to docker compose limitations)

### CODE REVIEW

File: lanl/docker-compose/ochami-krakend-ce.yml [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
-    image: devopsfaith/krakend
+    hostname: SI.krakend
+    container_name: SI.krakend
+    image: ghcr.io/openchami/krakend:acme
     environment:
       - KRAKEND_CONFIG=/etc/krakend/krakend.json
       - KRAKEND_PORT=8080
-    hostname: krakend-ce
+    healthcheck:
+      test: ["CMD-SHELL", "CURL_CA_BUNDLE=/root/.acme.sh/SI.krakend_ecc/SI.krakend.cer", "curl", "-f", "https:/SI.krakend:8080/__health"]
```
Comment (alexlovelltroy) [inline lanl/docker-compose/ochami-krakend-ce.yml:28]: Does this indicate that .acme.sh must be present in the /root directory of the VM?  The host machine?  The container?  I think a comment here to clarify which /root and why would be very helpful.

File: lanl/docker-compose/ochami-krakend-ce.yml [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
+      retries: 5
+      start_period: 10s
     depends_on:
       - bss
       - smd
+      - step-ca
     ports:
       - "8080:8080"
     networks:
       - internal
```
Comment (alexlovelltroy) [inline lanl/docker-compose/ochami-krakend-ce.yml:40]: I thought we were planning to separate the internal and external networks involved here so that Krakend-ce could be the only container accessible from outside the docker network.   Should that be part of this PR, or do we expect that in another PR?

File: lanl/docker-compose/ochami-krakend-ce.yml [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
     volumes:
-      - ./configs/krakend.json:/etc/krakend/krakend.json:ro
+      - ./configs/krakend.json:/etc/krakend.json:ro
+      - ./configs/step-ca/config/defaults.json:/mnt/defaults.json:ro
```
Comment (alexlovelltroy) [inline lanl/docker-compose/ochami-krakend-ce.yml:43]: is defaults.json intended to be present before launching the docker compose or will it be created by the step-ca container?  We should indicate either way with comment(s) in the file.

**Review by alexlovelltroy — CHANGES_REQUESTED** [review]
There are a few comments that need to be added and a question about internal versus external networks.

File: lanl/docker-compose/ochami-krakend-ce.yml (hunk 2 above)
Comment (njones-lanl) [reply lanl/docker-compose/ochami-krakend-ce.yml:40]: I think we want that to be part of another PR, at least for the CA I like it being externally accessible so we can have clients bootstrap the core certs.

File: lanl/docker-compose/ochami-krakend-ce.yml (hunk 3 above)
Comment (njones-lanl) [reply lanl/docker-compose/ochami-krakend-ce.yml:43]: It's auto generated by launching the step-ca container. I'll add it as a comment. In this case (the krakend container) the `defaults.json` contains the fingerprint that the step cli uses to bootstrap the core CA into the container. Will add a comment.

**Review by alexlovelltroy — APPROVED** [review]
LGTM
